packages/braintree-integration/src/braintree-paypal/braintree-paypal-payment-strategy.ts

```diff
@@ -257,11 +297,139 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
+                    );
+
+                    if (submitForm && typeof submitForm === 'function') {
+                        submitForm();
+                    }
+                },
+                onCancel: () => {
+                    this.toggleLoadingIndicator(false);
+                },
+                onError: (error: Error) => {
+                    this.toggleLoadingIndicator(false);
+
+                    onError?.(error);
+                },
+            });
+
+            if (onRenderButton && typeof onRenderButton === 'function') {
+                onRenderButton();
+            }
+
+            if (this.paypalButtonRender.isEligible()) {
+                this.paypalButtonRender.render(`${containerId}`);
+            }
+        } else {
+            this.removeElement(containerId.split('#')[1]);
+        }
+    }
+
+    private async setupPayment(
+        braintreePaypalCheckout: BraintreePaypalCheckout,
+        method: string,
+        onPaymentError: BraintreePaypalPaymentInitializeOptions['onPaymentError'],
+    ): Promise<string> {
+        const state = this.paymentIntegrationService.getState();
+
+        try {
+            const customer = state.getCustomer();
+
+            const paymentMethod: PaymentMethod = state.getPaymentMethodOrThrow(method);
+
+            const amount = state.getCheckoutOrThrow().outstandingBalance;
+            const currencyCode = state.getStoreConfigOrThrow().currency.code;
+
+            const shippingAddress = state.getShippingAddress();
+
+            const address = shippingAddress || customer?.addresses[0];
+
+            const shippingAddressOverride = address
+                ? mapToBraintreeShippingAddressOverride(address)
+                : undefined;
+
+            return await braintreePaypalCheckout.createPayment({
+                flow: 'checkout',
+                enableShippingAddress: true,
+                shippingAddressEditable: false,
+                shippingAddressOverride,
+                amount,
+                currency: currencyCode,
+                offerCredit: false,
+                intent: paymentMethod.initializationData.intent,
+            });
+        } catch (error) {
+            if (onPaymentError) {
+                onPaymentError(error);
+            }
+
+            throw error;
+        }
+    }
+
+    private async tokenizePayment(
+        authorizeData: PaypalAuthorizeData,
+        braintreePaypalCheckout: BraintreePaypalCheckout,
+        onError?: (error: BraintreeError | StandardError) => void,
+    ): Promise<BraintreeTokenizePayload> {
+        try {
+            return await braintreePaypalCheckout.tokenizePayment(authorizeData);
+        } catch (error) {
+            if (onError) {
+                onError(error);
+            }
 
-        if (this.braintreeHostWindow.paypal && isMessageContainerAvailable) {
+            throw error;
+        }
+    }
+
+    private renderPayPalMessages(initializationData?: any) {
```

Comment:
`any` type is forbidden for any new code in checkout sdk

Reply:
Made decision to not pass this parameter, moved it to higher level